homeassistant/components/climate/myicomfort.py

```diff
@@ -0,0 +1,225 @@
+        """Return device specific state attributes."""
+        return {
+        }
+
+    @property
+    def state(self):
+        """Return the current operational state."""
+        return SYSTEM_STATES[self._api.state]
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_UNITS[self._api.temperature_units]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self._min_temp:
+            return self._min_temp
+        return super().min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self._max_temp:
+            return self._max_temp
+        return super().max_temp
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._api.op_mode == 1:
+            return min(self._api.set_points)
+        if self._api.op_mode == 2:
+            return max(self._api.set_points)
+        return None
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._api.current_temperature
+
+    @property
+    def target_temperature_high(self):
+        """Return the highbound target temperature we try to reach."""
+        if self._api.op_mode == 3:
+            return max(self._api.set_points)
+        return None
+
+    @property
+    def target_temperature_low(self):
+        """Return the lowbound target temperature we try to reach."""
+        if self._api.op_mode == 3:
+            return min(self._api.set_points)
+        return None
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._api.current_humidity
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return OP_MODES[self._api.op_mode]
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OP_MODES
+
+    @property
+    def is_away_mode_on(self):
+        """Return the current away mode status."""
+        return self._api.away_mode
+
+    @property
+    def current_fan_mode(self):
+        """Return the current fan mode."""
+        return FAN_MODES[self._api.fan_mode]
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return FAN_MODES
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature. API expects a tuple."""
+        if not self._api.away_mode:
+            if kwargs.get(ATTR_TEMPERATURE) is not None:
+                self._api.set_points = (kwargs.get(ATTR_TEMPERATURE), )
```

Comment:
Use `kwargs[ATTR_TEMPERATURE]` since we know the key is in the dict.

Reply:
Please be sure the documentation for this new component is clear what models of Lennox iComfort thermostats it supports.    There are several different iComfort smart stats that are WiFi-enabled.   I suspect this component only support the lower model that is actually named "[iComfort WiFi Thermostat](https://www.lennox.com/products/comfort-controls/thermostats/icomfortwi-fi)" .   

My understanding is that this lower model uses a different API than the upper end[ iComfort series product line ](https://www.lennox.com/products/comfort-controls/thermostats)  (S30, E30, M30, 5500, 7500).

Many of us have been struggling to get a working custom component for the upper end models.   [Please see this discussion](https://community.home-assistant.io/t/lennox-icomfort-s30-e30-or-m30-thermostat-api/49257/13) for reference on the attempts that have been made and why it is important to make a distinction in the models this component will support.

Thanks!

Reply:
I am aware that the iComfort Wifi is the unique guy in the bunch and the others use a different cloud interface.  The docs already say 'iComfort WiFi' but I have added a line explicitly stating this component doesn't support the other iComfort thermostats.  Thanks for the feedback.